src/VecSim/spaces/spaces.h

```diff
@@ -30,4 +31,19 @@ static int inline is_little_endian() {
     return *(char *)&x;
 }
 
+static inline auto getCpuOptimizationFeatures(const void *arch_opt = nullptr) {
+#if defined(CPU_FEATURES_ARCH_X86_64)
+    using FeaturesType = cpu_features::X86Features;
+    constexpr auto getFeatures = cpu_features::GetX86Info;
+#elif defined(CPU_FEATURES_ARCH_AARCH64)
+    using FeaturesType = cpu_features::Aarch64Features;
+    constexpr auto getFeatures = cpu_features::GetAarch64Info;
+#else
+    using FeaturesType = cpu_features::X86Features; // Fallback
+    constexpr auto getFeatures = cpu_features::GetX86Info;
```

Comment:
exactly the same as `CPU_FEATURES_ARCH_X86_64`? Can we remove it/the specific `CPU_FEATURES_ARCH_X86_64` implementation?